doc/syntax.rst

```diff
@@ -289,6 +290,30 @@ as an array:
 
 * each row represents a shot/run
 
+For-loops
+~~~~~~~~~
+
+Similar to Python, for-loops can be declared using the ``for ... in ...`` syntax, followed by lines
+of indented statements. Notice that there is no ``:`` at the end of the for-statement. The for-loop
+variable type must be declared followed by either a list of values, of the specified type, or a
+range using the syntax ``from:to:step``.
+
+For example:
+
+.. code-block:: python
+
+    for int i in [0, 2, 1, 0, 2, 1]
+        MZgate(phases[i], phases[i+1]) | [i, i+1]
+
+where ``phases`` could be an array declare above, or:
+
+.. code-block:: python
+
+    for int m in 2:10:2
+        MeasureX | m
+
+measuring over modes 2, 4, 6 and 8.
+
```

Comment:
Can you also do
```python
for int i in phases
    MZgate(i, 0.5) | [0, 1]
```

Reply:
(Just another thought --- do nested loops work?)
```python
for int i in 0:5
    for int j in 0:i
        BSgate(0.5) | [i, j]
```

Reply:
I was thinking about this as well, and unfortunately, no, neither of those two currently work. That's why I wrote `followed by either a list of values` rather than an array for the for-loop statement. Both of these would be great to support though, and I don't think they would be that difficult to implement actually.

Reply:
Might be worth mentioning briefly that these _aren't_ supported :)